R/git.R

```diff
@@ -0,0 +1,84 @@
+#' Check if git repo is clean
+#'
+#' Check that git repo is up to date with remote by checking
+#' \code{git status --porcelian}. If import_dir is not a git root directory
+#' this will search parents of the import_dir for the git root
+#'
+#' @param import_dir The directory of the dettl import to check.
+#' @return True if repo is up to date. False otherwise.
+#'
+#' @keywords internal
+git_is_clean <- function(import_dir) {
```

Comment:
`import_dir` is is really `path`?

As indicated by the comment where this function is used, this function is probably better labelled `git_repo_is_clean` because it descends to the `.git` repo from the path given as an argument, rather than checking the status at the path that is given

Reply:
Yeah it is a path. Better naming is definitely more clear, I've updated that.